src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/RemoveAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Deletes a deployment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(bool))]
+    public class RemoveAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "RemoveByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "RemoveByDeploymentName";
+
+        /// <summary>
+        /// The input object parameter set.
+        /// </summary>
+        internal const string InputObjectParameterSet = "RemoveByInputObject";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("DeploymentId", "ResourceId")]
+        [Parameter(ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
```

Comment:
@Tiano2017 please add the `ValueFromPipelineByPropertyName=true` property to this parameter and remove it from the `-Name` parameter

Reply:
we actually want "Name" to be from pipeline, because the output of Get-AzureRmDeployment only contains "Name" but not "Id".

Reply:
@Tiano2017 the result of `Get-AzureRmDeployment` will be bound to the `-InputObject` parameter. We want to enable the scenario (as mentioned above for `Get-AzureRmDeployment`) where the user can pipe the result of `Get-AzureRmResource` to a cmdlet and have the `-ResourceId` bound, which is enable when you add the `ValueFromPipelineByPropertyName` property to this parameter and remove it from the `-Name` parameter.

Reply:
as mentioned in the other comment. Get-AzureRmResource doesn't return deployments. I can add ValueFromPipelineByPropertyName  for resourceId parameter, but I think we need to keep the on on Name parameter. 

---
In reply to: [207689305](https://github.com/Azure/azure-powershell/pull/6516#discussion_r207689305) [](ancestors = 207689305)

Reply:
this was addressed. 

---
In reply to: [205822057](https://github.com/Azure/azure-powershell/pull/6516#discussion_r205822057) [](ancestors = 205822057)